lib/serialize_attributes/store.rb

```diff
@@ -193,29 +169,52 @@ def initialize(original, store) # rubocop:disable Lint/MissingSuper
         @store = store
       end
 
+      def cast(value)
+        value = deserialize(value.stringify_keys) if value.is_a?(Hash)
+        value
```

Comment:
I found this hard to understand. Maybe this instead?

```suggestion
        return deserialize(value.stringify_keys) if value.is_a?(Hash)
        
        value
```

Reply:
Not a fan of either of these implementations now I look again 😄  I've switched to a `case` statement to make the Hash check more explicit (and ugly! make bad code ugly)